packages/react/src/components/DesignLibrary/models.ts

```diff
@@ -0,0 +1,85 @@
+import { ImportEntry, ImportEntryInfo } from '@sitecore-content-sdk/core/codegen';
+import {
+  ComponentFields,
+  ComponentParams,
+  ComponentRendering,
+  RouteData,
+} from '@sitecore-content-sdk/core/layout';
+import { DesignLibraryStatus } from '@sitecore-content-sdk/core/editing';
+import { Page } from '@sitecore-content-sdk/core/client';
+import { ComponentPreviewEventArgs } from '@sitecore-content-sdk/core/codegen';
+import { ComponentMap } from '../sharedTypes';
+
+export type ImportMapImport = {
+  default: ImportEntry[];
+};
+
+export type DynamicComponent = React.ComponentType<{
+  [key: string]: unknown;
+  fields: ComponentFields;
+  params: ComponentParams;
+}>;
+
+// @MAJOR-RELEASE-TODO - Make importMap required in next major version
+export type DesignLibraryProps = {
+  /**
+   * The dynamic import for import map to be used in variant generation mode.
+   * Currently it's optional but it will be required in the next major version.
+   */
+  loadImportMap?: () => Promise<ImportMapImport>;
+};
+
+export type DesignLibraryServerProps = {
+  /**
+   * Component Map will be used to map Sitecore component names to app implementation
+   * When rendered within a <SitecoreProvider> component, defaults to the context componentMap.
+   * When rendered as a server placeholder, this prop must be provided.
```

Comment:
Not sure if this description is truthy, looks like it was pulled from AppPlaceholder, in our case `componentMap` is required, right? Can we mark it as required?